pkg/storage/ec/client.go

```diff
@@ -274,14 +274,15 @@ func (ec *ecClient) Repair(ctx context.Context, limits []*pb.AddressedOrderLimit
 }
 
 func (ec *ecClient) putPiece(ctx, parent context.Context, limit *pb.AddressedOrderLimit, data io.ReadCloser, expiration time.Time) (hash *pb.PieceHash, err error) {
-	defer mon.Task()(&ctx, "node: ", limit.GetLimit().StorageNodeId.String()[0:8])(&err)
 	defer func() { err = errs.Combine(err, data.Close()) }()
 
 	if limit == nil {
 		_, _ = io.Copy(ioutil.Discard, data)
 		return nil, nil
 	}
 
+	defer mon.Task()(&ctx, "node: "+limit.GetLimit().StorageNodeId.String()[0:8])(&err)
```

Comment:
(A) I think it would still be worth keeping records on calls to this function and how long they take, even in the limit=nil case.
(B), moving this defer after the other means switching the order they get run, so errors from data.Close() would not be reported correctly

How about putting something like this at the top instead?

    nodeName := "nil"
    if limit != nil {
        nodeName = limit.GetLimit().StorageNodeId.String()[0:8]
    }
    defer mon.Task()(&ctx, "node: "+nodeName)(&err)

Reply:
:+1:

Reply:
I like that idea! I've updated it.